File: 35-SearchInsertPosition/35-SearchInsertPosition.java
// Last updated: 4/3/2026, 2:04:09 PM
class Solution {
    public int searchInsert(int[] nums, int target) {
        int begin=0, end=nums.length-1;
        while(begin<=end) {
            int mid=(begin+end)/2;
            if(nums[mid]<target) {
                begin=mid+1;
            }
            else if(nums[mid]>target) {
                end=mid-1;
            }
            else
                return mid;
        }
        return begin;
    }
}

File: 35-SearchInsertPosition/35-SearchInsertPosition.cpp
// Last updated: 4/3/2026, 2:04:04 PM
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int beg = -0, end = nums.size() - 1, ans = -1;
        if (nums[0] >= target)
            return 0;
        if (nums[nums.size() - 1] < target)
            return nums.size();
        while (beg <= end) {
            int mid = (beg + end) >> 1;
            if (nums[mid] >= target) {
                ans = mid;
                end = mid - 1;
            } else {
                beg = mid + 1;
            }
        }
        return ans;
    }
};